File: Solutions/74. Search a 2D Matrix/74. Search a 2D Matrix.py
class Solution:
    def searchMatrix(self, mat: List[List[int]], target: int) -> bool:
        m,n=len(mat),len(mat[0])
        l,r=0,m*n-1
        while l<=r:
            mid=(l+r)//2
            if mat[mid//n][mid%n]==target:
                return True
            if mat[mid//n][mid%n]<target:
                l=mid+1
            else:
                r=mid-1
        return False

File: Solutions/74. Search a 2D Matrix/74. Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& mat, int target) {
        int m=mat.size(),n=mat[0].size();
        int lo=0,hi=m*n-1;
        while(lo<=hi){
            int mid=lo+hi>>1;
            if(mat[mid/n][mid%n]==target){
                return true;
            }
            if(mat[mid/n][mid%n]<target){
                lo=mid+1;
            }else{
                hi=mid-1;
            }
        }
        return false;
    }
};